api/utils/reaction_collector.go

```diff
@@ -0,0 +1,66 @@
+package utils
+
+import (
+	"github.com/DisgoOrg/disgo/api"
+	"github.com/DisgoOrg/disgo/api/events"
+)
+
+// NewReactionCollector gives you a channel to receive on and a function to close the collector
+func NewReactionCollector(disgo api.Disgo, channelID api.Snowflake, guildID api.Snowflake, messageID api.Snowflake, filter ReactionFilter) (chan *api.MessageReaction, func()) {
+	ch := make(chan *api.MessageReaction)
+
+	col := &ReactionCollector{
+		Filter:    filter,
+		Channel:   ch,
+		ChannelID: channelID,
+		GuildID:   guildID,
+		MessageID: messageID,
+	}
+
+	cls := func() {
+		close(ch)
+		disgo.EventManager().RemoveEventListener(col)
+	}
+
+	col.Close = cls
+
+	disgo.EventManager().AddEventListeners(col)
+
+	return ch, cls
+}
+
+// NewReactionCollectorFromMessage is an overload of NewReactionCollector that takes an api.Message for information
+//goland:noinspection GoUnusedExportedFunction
+func NewReactionCollectorFromMessage(message *api.Message, filter ReactionFilter) (chan *api.MessageReaction, func()) {
+	return NewReactionCollector(message.Disgo, message.ChannelID, message.ID, *message.GuildID, filter)
+}
+
+// ReactionFilter used to filter api.MessageReaction in a ReactionCollector
+type ReactionFilter func(reaction *api.MessageReaction) bool
```

Comment:
I think that should take the event as MessageReaction has way too less data
```suggestion
type ReactionFilter func(reaction *events.MessageReactionAddEvent) bool
```